_posts/2023-07-17-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,153 @@
+
+Now, click on Realm roles and create two roles, user and admin. 
+
+Finally, create a user called `alice` whose `First name` is Alice and `Last name` is Smith and assign her the user and admin roles. Steps for assigning roles can be found in the https://www.keycloak.org/docs/latest/server_admin/#proc-assigning-role-mappings_server_administration_guide[ Keycloak documentation]. We will be using different scopes to query these pieces of information about Alice. 
+
+Next, navigate to the `credentials` tab and create a password for Alice. Toggle the `temporary` switch off so you are not prompted to update the password after the first login. The list of scope are the values Keycloak allows and it varies for different OpenID providers.
+
+
+Note: The realm and client names are case sensitive. 
+
+== Server Configuration
+
+First clone the `elytron-example` repo locally: 
+```
+    git clone https://github.com/wildfly-security-incubator/elytron-examples
+    cd elytron-examples
+```
+
+In order to authenticate a user using OpenID connect, we need to configure the server to use OpenID attributes. This can be done using one of the following configurations: 
+
+=== Deployment Configuration
+
+The configuration required to secure an application with OpenID Connect can be specified in the deployment using the `oidc.json` file.
+
+The first step is to create an `oidc.json` configuration file in the `WEB-INF` directory of the `simple-webapp-oidc` project. 
+
+Note: There is a `simple-webapp-oidc` project inside the `elytron-examples` directory. However, we will not be using that one. Instead, we will be using the slightly modified version of that inside the `elytron-oidc-client-scope` project. 
+
+There is already an `oidc.json` file inside the `WEB-INF` directory. So, we will only be editing that file by adding scope values to the existing deployment configuration. The following is the contents of the `oidc.json` file used in this example:
+```
+{
+    "client-id" : "myclient",
+    "provider-url" : "${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm",
+    "public-client" : "true",
+    "scope" : "profile email roles web-origins microprofile-jwt offline_access",
+    "principal-attribute" : "preferred_username",
+    "ssl-required" : "EXTERNAL"
+}
+```
+
+Note that we have specified the scope values to be `profile`, `email`, `web-origins`, `microprofile-jwt`, `roles` and `offline_access` in a space delimited list inside the `oidc.json` file. `profile`, `email` and `offline_access` are OpenID built-in scopes, while `web-origin`, `microprofile-jwt` and `roles` are Keycloak specific scope values and allow access to additional claims. Read the descriptions under the `client scope` tab for myclient to learn more about the functions of these scope values. 
+
+The next step is to set the auth-method to OIDC in the application’s `web.xml` file. This should have already been done, but if that is not the case, please add the following command: 
```

Comment:
`auth-method`
`OIDC`

Reply:
"This should have already been done, but if that is not the case, please add the following command: "

This is a bit confusing since this will always be present in the example file.